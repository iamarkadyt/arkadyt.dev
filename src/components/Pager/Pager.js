import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';

import Projects from './Pages/Projects/Projects'; // /projects route component.
import EmailMe from './Pages/EmailMe/EmailMe'; // /emailme route component.
import NavBar from './NavBar/NavBar';

import classes from './Pager.module.css';

/**
 * Holds root level router and global NavBar.
 */
const pager = () => {
    const routes = {
        '/projects': 'Projects',
        '/emailme': 'Email Me',
    }

    return (
        <Router>
            <div className={classes.container}>
                <NavBar routes={routes} />
                <Route path='/projects' component={Projects} />
                <Route path='/emailme' component={EmailMe} />
            </div>
        </Router>
    );
};

export default pager;