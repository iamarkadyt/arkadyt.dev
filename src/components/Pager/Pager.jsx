import React from 'react';
import { BrowserRouter as Router, Route, Redirect, Switch } from 'react-router-dom';

import Projects from './_Pages/Projects/Projects'; // /projects route component.
import EmailMe from './_Pages/EmailMe/EmailMe'; // /emailme route component.
import NavBar from './NavBar/NavBar';
import DotNav from './DotNav/DotNav';
import SMLinks from './SMLinks/SMLinks';

import classes from './Pager.module.css';
// import { connect } from 'react-redux';
import { GoChevronLeft, GoChevronRight } from 'react-icons/lib/go/';


const pager = () => {
    /* 'route' : 'nav-button-name' */
    const routes = {
        '/projects': 'Projects',
        '/emailme': 'Email Me',
    }

    return <Router basename='/pf-webapp'>
        <div className={classes.container}>
            <div className={classes.navtop}>
                <NavBar routes={routes} />
            </div>
            <a href='' className={classes.arrowLeft}><GoChevronLeft /></a>
            <Switch>
                <Route path='/projects' component={Projects} />
                <Route path='/emailme' component={EmailMe} />
                <Redirect to='/emailme' />
            </Switch>
            <a href='' className={classes.arrowRight}><GoChevronRight /></a>
            <div className={classes.navbottom}>
                <Switch>
                    <Route path='/projects' render={
                        () => <DotNav urls={['1', '2']} />} />
                    <Route path='/emailme' component={SMLinks} />
                </Switch>
            </div>
        </div>
    </Router>;
};

// const mapStateToProps = state => {
//     return {
//         projectsRoutes: state.projectsRoutes
//     }
// }
export default pager;