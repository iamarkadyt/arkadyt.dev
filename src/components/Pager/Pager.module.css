.container {
    display: grid;
    grid-template-columns: 50px auto;
    width: 100vw;
    height: 100vh;
    background-size: cover;
}

img.background {
    position: fixed;
    z-index: -50;
    top: -3%;
    left: -3%;
    width: 106%;
    height: 106%;
    object-fit: cover;
    filter: blur(5px) brightness(50%);
}

.arrowLeft,
.arrowRight {
    grid-row: 2/3;
    display: flex;
    justify-content: center;
    align-items: center;
    color: #ffffff;
    font-size: 2em;
}

.arrowLeft {
    grid-column: 1/2;
}

.arrowRight {
    grid-column: 3/4;
}

.navtop,
.navbottom {
    grid-column: 1/4;
    align-self: center;
    
    display: flex;
    justify-content: center;
    align-content: center;
}

.navtop {
    grid-row: 1/2;
}

.navbottom {
    grid-row: 3/4;
}