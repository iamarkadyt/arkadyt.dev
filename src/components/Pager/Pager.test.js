import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Pager } from './Pager';

configure({ adapter: new Adapter() });

it('matches snapshot.', () => {
    const tree = shallow(<Pager projectRoutes={['1', '11']} componentLoaded={true} />);
    expect(tree).toMatchSnapshot();
});