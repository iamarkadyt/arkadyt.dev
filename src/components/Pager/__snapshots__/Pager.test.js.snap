// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot. 1`] = `
<BrowserRouter
  basename="/pf-webapp"
>
  <div
    className="container"
  >
    <div
      className="navtop"
    >
      <navBar
        routes={
          Object {
            "/emailme": "Email Me",
            "/projects": "Projects",
          }
        }
      />
    </div>
    <Switch>
      <Route
        path="/projects"
        render={[Function]}
      />
      <Route
        path="/emailme"
        render={[Function]}
      />
      <Redirect
        push={false}
        to="/emailme"
      />
    </Switch>
    <div
      className="navbottom"
    >
      <Switch>
        <Route
          path="/projects"
          render={[Function]}
        />
        <Route
          component={[Function]}
          path="/emailme"
        />
      </Switch>
    </div>
  </div>
</BrowserRouter>
`;
